src/main/java/com/hubspot/jinjava/el/ext/AstDict.java

```diff
@@ -44,6 +45,11 @@ public Object eval(Bindings bindings, ELContext context) {
         } else {
           key = ((AstIdentifier) entryKey).getName();
         }
+      } else if (entryKey instanceof AstNumber) {
+        // This is a hack to treat numeric keys as string keys in the dictionary.
+        // In most cases this is adequate since the keys are typically treated as
+        // strings.
+        key = entryKey.eval(bindings, context).toString();
       } else {
         throw new TemplateStateException(
           "Dict key must be a string or identifier, was: " + entryKey
```

Comment:
It's good to update the error message here to include the number as a key possibility, what do you think?
```suggestion
          "Dict key must be a string, or identifier, or a  number, was: " + entryKey
```